networkx/algorithms/approximation/steinertree.py

```diff
@@ -0,0 +1,76 @@
+import networkx as nx
+from itertools import combinations, chain
+from networkx.utils import pairwise, not_implemented_for
+
+__all__ = ['metric_closure', 'steiner_tree']
+
+
+@not_implemented_for('directed')
```

Comment:
I think this algorithm should work for directed graphs too, as long as you use `permutations` instead of `combinations`. If you feel up to it, you can make the change to the code and add a unit test for a directed graph.